### PR TITLE
Modified Github constructor, getCommits, and getCommit. Added getComments, createComment, and updateComment.

### PR DESCRIPTION
**Pull request by lukeburns**
### Modifications:
- options parameter for the Github constructor is now optional, so you can instantiate with `new Github()` to access public data without specifying authorization information.
- `getCommits` was failing when only passed a callback. options parameter is now optional.
- corrected path (should not contain '/git/') and removed unnecessary argument `branch` from `getCommit(sha, cb)`.
- I removed the `branch` parameter from the `getCommit` test, because the Github API does not take a `branch` parameter.

### Added:
- Get commit comments: `getComments([sha,] cb)`
- Create a commit comment: `createComment(sha, body, [options,] cb)`
- Update a commit comment: `updateComment(id, body, cb)`
- Added tests for `getCommits` and `getComments`

### CODE REVIEW

**Comment by aendrew** [comment]
Apologies for the delay in getting to what looks like a really good PR! Will try to review this week.

**Comment by aendrew** [comment]
Hi! Apologies for the delay! Mind adding tests for createComment and updateComment? Adding a deleteComment method might help for cleaning up the test repo afterwards. Thanks!

**Comment by adjohnson916** [comment]
+1 for comments API. If this is a good implementation, any chance of it getting merged?

**Comment by lukeburns** [comment]
I don't think I'll be able to get a chance to add the requested tests until late October during fall break.

File: github.js [hunk 1]
```diff
@@ -683,7 +681,10 @@
       // -------
 
       this.getCommits = function(options, cb) {
-          options = options || {};
+          if(arguments.length === 1 && typeof arguments[0] === "function") {
```
Comment (AurelioDeRosa) [inline github.js]: This can be rewritten as:

```js
if(arguments.length === 1 && typeof options === "function") {
```

**Comment by aendrew** [comment]
@adjohnson916 Not a problem! We're pushing for a 0.10.7 release either end of the month or mid-November (depending on how things go), then 0.10.8 sometime next year; happy to merge this for either release! Would love to get it in!

**Comment by aendrew** [comment]
@lukeburns Not a problem! We're pushing for a 0.10.7 release either end of the month or mid-November (depending on how things go), then 0.10.8 sometime next year; happy to merge this for either release! Would love to get it in!

**Comment by aendrew** [comment]
Closed in error; reopening.

**Comment by AurelioDeRosa** [comment]
Hi @lukeburns.  Are you still interested in updating this PR?

**Comment by AurelioDeRosa** [comment]
Most of the methods added in this PR have been added to the library and it needs a strong rebase (e.g. we have changed the testing library). Closing it.

**Pull request closed without merging**